common/src/python/outputs/errors.py

```diff
@@ -25,46 +20,16 @@ class JSONLocation(BaseModel):
 class FileError(BaseModel):
     """Represents an error that might be found in file during a step in a
     pipeline."""
-    error_type: ErrorType
+    error_type: Literal['alert', 'error'] = Field(serialization_alias='type')
+    error_code: str = Field(serialization_alias='code')
     error_id: Optional[str] = None
     error_location: Optional[CSVLocation | JSONLocation] = None
     container_id: Optional[str] = None
+    flywheel_path: Optional[str] = None
     value: Optional[str] = None
     expected: Optional[str] = None
     message: str
 
-    # pylint: disable=no-self-use
-    @field_serializer('error_location')
-    def serialize_location(self,
-                           location: Optional[CSVLocation | JSONLocation]):
-        """Serializes the error_location field to a JSON string.
-
-        Args:
-          location: the location object
-        Returns:
-          JSON string representation of the location object
-        """
-        if not location:
-            return None
-
-        return location.model_dump_json()
-

```

Comment:
don't we still need serialization for error_location?

Reply:
No, since this will be stored in the metadata, it can be an object rather than a string